src/replication.c

```diff
@@ -2636,9 +2635,12 @@ void replicationSetMaster(char *ip, int port) {
     /* Update oom_score_adj */
     setOOMScoreAdj(-1);
 
-    /* Force our slaves to resync with us as well. They may hopefully be able
-     * to partially resync with us, but we can notify the replid change. */
-    disconnectSlaves();
+    /* Here we don't disconnect with replicas, since they may hopefully be able
+     * to partially resync with us. We will disconnect with replicas and force
+     * them resync with us when changing replid on partially resync with new
+     * master, or finishing transferring RDB and preparing loading DB on full
+     * sync with new master. */
+
```

Comment:
@oranagra here, i think we don't need to disconnect with replicas, as comments above, did i miss somethings?

Reply:
i suppose that's right.
i.e. the case of being given a new master, is now similar to the case of re-connecting to our existing master and discovering a new replid.
by "now" i mean since psync2.

@yossigo @soloestoy @madolson please ack and commend if you see anything wrong with this PR.